commands/root.go

```diff
@@ -0,0 +1,166 @@
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/platformsh/cli/internal"
+	"github.com/platformsh/cli/internal/legacy"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func init() {
+	RootCmd.PersistentFlags().BoolP("version", "V", false, "Displays the Platform.sh CLI version")
+	RootCmd.PersistentFlags().String("phar-path", "", "Uses a local .phar file for the Legacy Platform.sh CLI")
+	RootCmd.PersistentFlags().Bool("debug", false, "Enable debug logging")
+
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.AutomaticEnv()
+	//nolint:errcheck
+	viper.BindPFlags(RootCmd.PersistentFlags())
+	viper.SetEnvPrefix("platformsh_cli")
+	log.SetOutput(color.Error)
+}
+
+var (
+	shellConfigSnippet = regexp.MustCompile("# BEGIN SNIPPET: Platform.sh CLI configuration(?s).+?# END SNIPPET")
+	updateMessageChan  = make(chan *internal.ReleaseInfo, 1)
+)
+
+var RootCmd = &cobra.Command{
+	Use:                "platform",
+	Short:              "Platform.sh CLI",
+	Long:               `The Platform.sh CLI is a command line interface for Platform.sh.`,
+	Args:               cobra.ArbitraryArgs,
+	FParseErrWhitelist: cobra.FParseErrWhitelist{UnknownFlags: true},
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		if viper.GetBool("version") {
+			VersionCmd.Run(cmd, []string{})
+			os.Exit(0)
+		}
+		go func() {
+			rel, _ := internal.CheckForUpdate("platformsh/cli", version)
+			updateMessageChan <- rel
+		}()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		c := &legacy.LegacyCLIWrapper{
+			Version:          version,
+			CustomPshCliPath: viper.GetString("phar-path"),
+			Debug:            viper.GetBool("debug"),
+		}
+		if err := c.Init(); err != nil {
+			debugLog("%s\n", color.RedString(err.Error()))
+			os.Exit(1)
+			return
+		}
+
+		exitCode := 0
+		defer func() {
+		}()
```

Comment:
empty function 🤔

Reply:
Good catch, it was needed to get the exit code but I rewrote that afterwards and it was left out :/